hedera-node/src/main/java/com/hedera/services/state/migration/MigrationRecordsManager.java

```diff
@@ -131,24 +165,39 @@ public void publishMigrationRecords(final Instant now) {
                     num -> publishSyntheticCreationForStakingFund(num, implicitAutoRenewPeriod));
         } else {
             // Publish free auto-renewal migration records if expiry is just being enabled
-            if (expiryJustEnabled) {
+            if (expiryJustEnabled && !areAllMigrationsSansTraceabilityFinished) {
                 publishContractFreeAutoRenewalRecords();
             }
         }
 
         // And we always publish records for any treasury clones that needed to be created
-        treasuryCloner
-                .getClonesCreated()
-                .forEach(
-                        account ->
-                                publishSyntheticCreation(
-                                        account.getKey(),
-                                        account.getExpiry() - now.getEpochSecond(),
-                                        asKeyUnchecked(account.getAccountKey()),
-                                        account.getMemo(),
-                                        TREASURY_CLONE_MEMO,
-                                        "treasury clone"));
+        if (!areAllMigrationsSansTraceabilityFinished) {
+            treasuryCloner
+                    .getClonesCreated()
+                    .forEach(
+                            account ->
+                                    publishSyntheticCreation(
+                                            account.getKey(),
+                                            account.getExpiry() - now.getEpochSecond(),
+                                            asKeyUnchecked(account.getAccountKey()),
+                                            account.getMemo(),
+                                            TREASURY_CLONE_MEMO,
+                                            "treasury clone"));
+        }
 
+        if (globalDynamicProperties.isTraceabilityMigrationEnabled()) {
+            // if we cannot perform the traceability migration with this txn
+            // do not mark migration records as streamed in the context and indicate to {@link
+            // ServicesTxnManager} to call this method until migration is finished
+            if (isSidecarGeneratingFunction(transactionContext.accessor().getFunction())) {
+                areAllMigrationsSansTraceabilityFinished = true;
+                return;
+            }
+            publishTraceabilityMigrationRecords();
+        }
+        // if we reach this line, traceability migration was given opportunity to execute,
+        // so we mark migration record streaming as completed
+        areAllMigrationsSansTraceabilityFinished = false;
```

Comment:
⚠️ It's very, very unlikely...but with this implementation it's _possible_ for a node to reconnect immediately after start-up; after all other nodes have streamed non-traceability records but **NOT** traceability records. (If the first transaction after the upgrade is a contract op.)

The reconnecting node will then incorrectly stream non-traceability records and get an ISS, because only information about completing **traceability** records is in state (`areMigrationRecordsStreamed`)

So I'd suggest continuing to use `areMigrationRecordsStreamed` as information about non-traceability records. Then we can have another flag, not in state, that controls streaming traceability records:
```
boolean areTraceabilityRecordsStreamed = false;
```

In `ServicesState.init()`, on a `RECONNECT` trigger, we can then call e.g.,
```
app.migrationRecordsManager().markTraceabilityRecordsStreamed();
```
to make sure any reconnecting node that missed the post-upgrade window will not stream traceability records.

Reply:
Makes perfect sense, fixed now